### PR TITLE
feat: Add silence and unsilence public api endpoint

### PR DESCRIPTION
**Pull request by ravishankar15**
# What this PR does
Exposes alert group silence and unsilence via public api endpoint

## Which issue(s) this PR closes
[#5026](https://github.com/grafana/oncall/issues/5026)

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] Added the relevant release notes label (see labels prefixed w/ `release:`). These labels dictate how your PR will
    show up in the autogenerated release notes.

### CODE REVIEW

File: docs/sources/oncall-api-reference/alertgroups.md [hunk 1]
```diff
@@ -130,6 +130,34 @@ curl "{{API_URL}}/api/v1/alert_groups/I68T24C13IFW1/unresolve" \
...
+
+**HTTP request**
+
+`POST {{API_URL}}/api/v1/alert_groups/<ALERT_GROUP_ID>/silence`
+
+| Parameter | Required | Description                                                                                                                                                                                                                                                                                                                                             |
+|-----------|:--------:|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `delay`    |    Yes    | The duration of silence in seconds, `-1` for silencing the alert forever |
+
+## unsilence an alert group
```
Comment (matiasb) [inline docs/sources/oncall-api-reference/alertgroups.md:149]: s/unsilence/Unsilence

File: engine/apps/public_api/tests/test_alert_groups.py [hunk 2]
```diff
@@ -635,3 +635,96 @@ def test_alert_group_unresolve(
...
+        root_alert_group=root_alert_group if attached else None,
+    )
+
+    client = APIClient()
+    url = reverse("api-public:alert_groups-unsilence", kwargs={"pk": alert_group.public_primary_key})
+    response = client.post(url, HTTP_AUTHORIZATION=token)
+
+    if status_code == status.HTTP_200_OK:
+        alert_group.refresh_from_db()
+        assert alert_group.silenced is False
```
Comment (matiasb) [inline engine/apps/public_api/tests/test_alert_groups.py:726]: I would also check the final state of the alert group after unsilence.

**Review by matiasb — COMMENTED** [review]
LGTM, minor comments added.

**Comment by ravishankar15** [comment]
I have made the changes

File: docs/sources/oncall-api-reference/alertgroups.md [hunk 3]
```diff
@@ -130,6 +130,34 @@ curl "{{API_URL}}/api/v1/alert_groups/I68T24C13IFW1/unresolve" \
 
 `POST {{API_URL}}/api/v1/alert_groups/<ALERT_GROUP_ID>/unresolve`
 
+## Silence an alert group
+
+```shell
+curl "{{API_URL}}/api/v1/alert_groups/I68T24C13IFW1/silence" \
+  --request POST \
+  --header "Authorization: meowmeowmeow"
+```
```
Comment (matiasb) [inline docs/sources/oncall-api-reference/alertgroups.md:139]: This curl call is missing the delay param, right?

**Review by matiasb — COMMENTED** [review]
One last comment, other than that, lgtm. Will be merging after the update.
Thanks for the contribution!

File: docs/sources/oncall-api-reference/alertgroups.md (hunk 3 above)
Comment (ravishankar15) [reply docs/sources/oncall-api-reference/alertgroups.md:143]: yup I have Made the changes

**Review by matiasb — APPROVED** [review]

**Pull request merged**